catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java

```diff
@@ -188,34 +188,36 @@ private boolean hasValidLocation(String location) {
   private static Optional<URI> getOriginalDerivedResourceUri(final Metacard metacard) {
     final Attribute derivedResourceUriAttribute =
         metacard.getAttribute(Metacard.DERIVED_RESOURCE_URI);
-    if (derivedResourceUriAttribute != null) {
-      for (Serializable value : derivedResourceUriAttribute.getValues()) {
-        if (value instanceof String) {
-          final String derivedResourceUriString = (String) value;
-          try {
-            final URI derivedResourceUri = new URI(derivedResourceUriString);
-
-            if (canBeChippedLocally(derivedResourceUri)) {
-              if (StringUtils.equals(ORIGINAL_QUALIFIER, derivedResourceUri.getFragment())) {
-                return Optional.of(derivedResourceUri);
-              }
-            } else {
-              for (NameValuePair parameter :
-                  URLEncodedUtils.parse(derivedResourceUri, StandardCharsets.UTF_8.name())) {
-                if (QUALIFIER_KEY.equals(parameter.getName())) {
-                  if (StringUtils.equals(ORIGINAL_QUALIFIER, parameter.getValue())) {
-                    return Optional.of(derivedResourceUri);
-                  }
-                }
-              }
-            }
-          } catch (URISyntaxException e) {
-            // ignore
+    if (derivedResourceUriAttribute == null) {
+      return Optional.empty();
+    }
+    List<String> derivedResourceUriStrings =
+        derivedResourceUriAttribute
+            .getValues()
+            .stream()
+            .filter(String.class::isInstance)
+            .map(Object::toString)
+            .collect(Collectors.toList());
+
+    for (String resourceUri : derivedResourceUriStrings) {
+      try {
+        final URI derivedResourceUri = new URI(resourceUri);
+
+        if (canBeChippedLocally(derivedResourceUri)
+            && StringUtils.equals(ORIGINAL_QUALIFIER, derivedResourceUri.getFragment())) {
+          return Optional.of(derivedResourceUri);
+        }
+        for (NameValuePair parameter :
+            URLEncodedUtils.parse(derivedResourceUri, StandardCharsets.UTF_8.name())) {
+          if (QUALIFIER_KEY.equals(parameter.getName())
+              && StringUtils.equals(ORIGINAL_QUALIFIER, parameter.getValue())) {
+            return Optional.of(derivedResourceUri);
           }
         }
+      } catch (URISyntaxException e) {
+        // ignore
```

Comment:
✏️  Are we sure we don't want to log it... even at trace or something?

Reply:
This is not an unexpected exception. If we get a `URISyntaxException` here, it means that the `resource.derived-uri` attribute value does not match the usual Alliance download URL format, so we don't have enough information to construct a chipping URL. It might be worthwhile to add a log statement just before `return Optional.empty();` so that all situations that drop down to that line will log the same thing.

Reply:
Will update to `This is not an unexpected exception, there is not enough info to construct a chipping URL`